otherlibs/stdune/src/ansi_color.ml

```diff
@@ -34,54 +36,115 @@ module Style = struct
     | `Bg_bright_magenta
     | `Bg_bright_cyan
     | `Bg_bright_white
+    | `Bg_8_bit_color of char
+    | `Bg_24_bit_color of char * char * char
     | `Bold
     | `Dim
     | `Italic
     | `Underline
     ]
 
-  let to_ansi_code = function
-    | `Fg_default -> 39
-    | `Fg_black -> 30
-    | `Fg_red -> 31
-    | `Fg_green -> 32
-    | `Fg_yellow -> 33
-    | `Fg_blue -> 34
-    | `Fg_magenta -> 35
-    | `Fg_cyan -> 36
-    | `Fg_white -> 37
-    | `Fg_bright_black -> 90
-    | `Fg_bright_red -> 91
-    | `Fg_bright_green -> 92
-    | `Fg_bright_yellow -> 93
-    | `Fg_bright_blue -> 94
-    | `Fg_bright_magenta -> 95
-    | `Fg_bright_cyan -> 96
-    | `Fg_bright_white -> 97
-    | `Bg_default -> 49
-    | `Bg_black -> 40
-    | `Bg_red -> 41
-    | `Bg_green -> 42
-    | `Bg_yellow -> 43
-    | `Bg_blue -> 44
-    | `Bg_magenta -> 45
-    | `Bg_cyan -> 46
-    | `Bg_white -> 47
-    | `Bg_bright_black -> 100
-    | `Bg_bright_red -> 101
-    | `Bg_bright_green -> 102
-    | `Bg_bright_yellow -> 103
-    | `Bg_bright_blue -> 104
-    | `Bg_bright_magenta -> 105
-    | `Bg_bright_cyan -> 106
-    | `Bg_bright_white -> 107
-    | `Bold -> 1
-    | `Dim -> 2
-    | `Italic -> 3
-    | `Underline -> 4
+  let to_ansi_code : t -> int list = function
```

Comment:
The intermediate list doesn't seem very useful

Reply:
It's not useful in most cases, but for 8 and 24 bit colors, several codes are needed at once which makes it tricky to represent otherwise.

Reply:
What do we do with this list? Just iterate over it to write it to a buffer and then immediately throw it away. Why not just write to a buffer directly then.